Lib/test/test_copy.py

```diff
@@ -881,6 +881,27 @@ def m(self):
         self.assertIs(g.b.__self__, g)
         g.b()
 
+    def test_deepcopy_custom_getattr_recursion_limit_exceeded(self):
+        class Proxy(object):
+            def __init__(self, proxied_object):
+                self.proxied_object = proxied_object
+
+            def __getattr__(self, name):
+                return getattr(self.proxied_object, name)
```

Comment:
`self.proxied_object` here causes a non-obvious recursion back to `__getattr__` when the object is being recreated with `deepcopy` (which uses `Proxy.__new__` and then looks for the `__setstate__` method).